starter-code/src/memory.js

```diff
@@ -1,12 +1,52 @@
 var MemoryGame = function (cards) {
   this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
 };
 
 MemoryGame.prototype.shuffleCards = function () {
+    var copy = [], n = this.cards.length, i;
+  
+    // While there remain elements to shuffle…
+    while (n) {
+  
+      // Pick a remaining element…
+      i = Math.floor(Math.random() * n--);
+
+      // And move it to the new array.
+      copy.push(this.cards.splice(i, 1)[0]);
+    }
+    return copy;
 };
 
 MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
-}
+  this.pairsClicked++;
+  if (firstCard === secondCard){
+    this.pairsGuessed++;
+    
+    $('#pairs_guessed').html(this.pairsGuessed);
+    this.pickedCards = [];
+  }
+  else { 
+  
```

Comment:
try to fix this, because if you click faster than the timer is being set to, your game is not working